firebase-firestore/src/main/java/com/google/firebase/firestore/bundle/BundleSerializer.java

```diff
@@ -307,11 +323,92 @@ private void decodeGeoPoint(Value.Builder builder, JSONObject geoPoint) {
             .setLongitude(geoPoint.optDouble("longitude", 0.0)));
   }
 
-  private void decodeTimestamp(Value.Builder builder, JSONObject timestamp) {
+  private Timestamp decodeTimestamp(String timestamp) {
+    // This method is copied from com.google.protobuf.util.Timestamps. See:
+    // https://chromium.googlesource.com/external/github.com/google/protobuf/+/HEAD/java/util/src/main/java/com/google/protobuf/util/Timestamps.java?autodive=0%2F#232
+
+    try {
+      int dayOffset = timestamp.indexOf('T');
+      if (dayOffset == -1) {
+        throw new IllegalArgumentException(
+            "Failed to parse timestamp: invalid timestamp \"" + timestamp + "\"");
```

Comment:
Optional: you could simplify this message to just "Invalid timestamp ...".

Reply:
Done